diff.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+
+def compare_source_files_objects(obj1, obj2):
+    diff_funcs = []
+    if obj1['name'] == obj2['name']:
+        if obj1['functions'] != obj2['functions'] or obj1['coverage'] != obj2['coverage']:
+            if obj1['functions'] == obj2['functions']:
+                obj1['functions'] = []
+            if obj1['functions'] != obj2['functions']:
+                for func1 in obj1['functions']:
+                    for func2 in obj2['functions']:
+                        if func1['name'] == func2['name']:
+                            if func1['exec'] is False and func2['exec'] is True:
+                                diff_funcs.append(func2)
+                obj1['functions'] = diff_funcs
+            list_val = []
+            for k, m in zip(obj1['coverage'], obj2['coverage']):
+                if m is None:
+                    k = None
```

Comment:
When I said `i` was confusing, it was because it's better not to re-use the same name as before.
You can keep i and j for the loop, but then use another name for the diff count (if you use `i` for the first and `j` for the second, use `k` for the result).